cs/src/core/Device/Devices.cs

```diff
@@ -24,7 +24,7 @@ public static class Devices
         /// <param name="capacity">The maximal number of bytes this storage device can accommondate, or CAPACITY_UNSPECIFIED if there is no such limit</param>
         /// <param name="recoverDevice">Whether to recover device metadata from existing files</param>
         /// <returns>Device instance</returns>
-        public static IDevice CreateLogDevice(string logPath, bool preallocateFile = true, bool deleteOnClose = false, long capacity = CAPACITY_UNSPECIFIED, bool recoverDevice = false)
+        public static IDevice CreateLogDevice(string logPath, bool preallocateFile = false, bool deleteOnClose = false, long capacity = CAPACITY_UNSPECIFIED, bool recoverDevice = false)
```

Comment:
Breaking change.

Reply:
Yes, it is breaking, but the benefit is that users do not create huge files when they use the simplest API. They opt in to the preallocation. Your example did that, and it seemed wasteful, so we thought it might be better to keep preallocation as opt-in.